src/metadataGeneration/typeResolver.ts

```diff
@@ -207,6 +207,16 @@ export class TypeResolver {
       }
     }
 
+    if (ts.isTypeOperatorNode(this.typeNode) && this.typeNode.operator === ts.SyntaxKind.KeyOfKeyword) {
+      const type = this.current.typeChecker.getTypeFromTypeNode(this.typeNode);
+      const indexedTypeName = this.current.typeChecker.typeToTypeNode(type);
```

Comment:
```suggestion
      const indexedTypeName =  const indexedTypeName = this.current.typeChecker.typeToString(this.current.typeChecker.getTypeFromTypeNode(this.typeNode.type))
;
```

Move this into the catch block, we don't need the name of the type we are getting the keys from unless we throw.
In this case however, we want to include the name of the type we perform the keyof on.

Reply:
Right sorry, should be fixed